pipeline/lib/custom_dockerfile.ml

```diff
@@ -50,92 +50,40 @@ let get_all_files ~repository =
   |> let> commit = Repository.src repository in
      Cache.get () { Get_files.Key.commit }
 
-let base_dockerfile ~base =
+let weekly = Current_cache.Schedule.v ~valid_for:(Duration.of_day 7) ()
+
+let base = Docker.pull ~schedule:weekly "ocaml/opam"
+
+let dockerfile ~base =
   let open Dockerfile in
   from (Docker.Image.hash base)
+  @@ run "sudo apt-get update"
   @@ run
-       "sudo apt-get update && sudo apt-get install -qq -yy libffi-dev \
-        liblmdb-dev m4 pkg-config gnuplot-x11 libgmp-dev libssl-dev \
-        libpcre3-dev && opam remote add origin https://opam.ocaml.org && opam \
-        update"
+       "sudo apt-get install -qq -yy --no-install-recommends libffi-dev \
+        liblmdb-dev m4 pkg-config libgmp-dev libssl-dev libpcre3-dev"
   @@ run "sudo mv /usr/bin/opam-2.1 /usr/bin/opam"
-
-let add_workdir =
-  let open Dockerfile in
-  copy ~src:[ "--chown=opam:opam ." ] ~dst:"bench-dir" ()
+  @@ run "opam remote add origin https://opam.ocaml.org"
+  @@ run "opam update"
   @@ workdir "bench-dir"
-  @@ add ~src:[ "--chown=opam ." ] ~dst:"." ()
+  @@ copy ~src:[ "--chown=opam:opam ./*.opam" ] ~dst:"." ()
   @@ run "opam exec -- opam pin -y -n --with-version=dev ."
+  @@ run "opam exec -- opam install -y --depext-only ."
+  @@ run "opam exec -- opam install -y --deps-only ."
+  @@ copy ~src:[ "--chown=opam:opam ." ] ~dst:"bench-dir" ()
+  @@ add ~src:[ "--chown=opam ." ] ~dst:"." ()
```

Comment:
Should this also be `--chown=opam:opam`? :)

Reply:
Yes indeed!